modules/tensor_mechanics/include/materials/ComputeCappedWeakPlaneStress.h

```diff
@@ -0,0 +1,357 @@
+
+  /** Computes the values of the yield functions, given stress and intnl, and puts them into _all_q
+   * @param p stress_zz
+   * @param q sqrt(stress_zx^2 + stress_zy^2)
+   * @param intnl The internal parameters (intnl[0] is shear, intnl[1] is tensile)
+   */
+  virtual void unsmoothedYieldFunctions(Real p, Real q, const std::vector<Real> & intnl);
+
+  /** Computes d(yieldFunctions)/d(p, q), and puts them into _all_q
+   * @param p stress_zz
+   * @param q sqrt(stress_zx^2 + stress_zy^2)
+   * @param intnl The internal parameters (intnl[0] is shear, intnl[1] is tensile)
+   */
+  virtual void dyieldFunctions(Real p, Real q, const std::vector<Real> & intnl);
+
+  /** Computes d(yieldFunctions)/d(intnl), and puts them into _all_q
+   * @param p stress_zz
+   * @param q sqrt(stress_zx^2 + stress_zy^2)
+   * @param intnl The internal parameters (intnl[0] is shear, intnl[1] is tensile)
+   */
+  virtual void dyieldFunctions_di(Real p, Real q, const std::vector<Real> & intnl);
+
+  /** Computes d(flowPotential)/d(p, q), and puts them into _all_q
+   * @param p stress_zz
+   * @param q sqrt(stress_zx^2 + stress_zy^2)
+   * @param intnl The internal parameters (intnl[0] is shear, intnl[1] is tensile)
+   */
+  virtual void dflowPotential(Real p, Real q, const std::vector<Real> & intnl);
+
+  /** Computes d2(flowPotential)/d(p, q)/d(p, q), and puts them into _all_q
+   * @param p stress_zz
+   * @param q sqrt(stress_zx^2 + stress_zy^2)
+   * @param intnl The internal parameters (intnl[0] is shear, intnl[1] is tensile)
+   */
+  virtual void d2flowPotential(Real p, Real q, const std::vector<Real> & intnl);
+
+  /** Computes d2(flowPotential)/d(p, q)/dintnl, and puts them into _all_q
+   * @param p stress_zz
+   * @param q sqrt(stress_zx^2 + stress_zy^2)
+   * @param intnl The internal parameters (intnl[0] is shear, intnl[1] is tensile)
+   */
+  virtual void d2flowPotential_di(Real p, Real q, const std::vector<Real> & intnl);
+
+  /** Computes the smoothed yield function
+   * @param p stress_zz
+   * @param q sqrt(stress_zx^2 + stress_zy^2)
+   * @param intnl The internal parameters (intnl[0] is shear, intnl[1] is tensile)
+   * @return The smoothed yield function value
+   */
+  virtual Real yieldF(Real p, Real q, const std::vector<Real> & intnl) const;
+
+  /** Smooths yield functions
+   */
+  Real ismoother(Real f_diff) const;
+
+  /** Derivative of ismoother
+   */
+  Real smoother(Real f_diff) const;
+
+  /** Derivative of smoother
+   */
+  Real dsmoother(Real f_diff) const;
+
+  /** Performs the return-map algorithm
+   */
+  virtual void returnMap();
+
+  struct f_and_derivs
+  {
+    Real f;
+    std::vector<Real> df;
+    std::vector<Real> df_di;
+    std::vector<Real> dg;
+    std::vector<std::vector<Real> > d2g;
+    std::vector<std::vector<Real> > d2g_di;
+
+    f_and_derivs():
+      f(0.0),
+      df(0),
+      df_di(0),
+      dg(0),
+      d2g(0),
+      d2g_di(0)
+    {}
+
+    f_and_derivs(unsigned num_var, unsigned num_intnl):
```

Comment:
Minor: Consider using the new delegating constructors available in C++11. This removes the need to duplicate your initialization lists. A very nice feature.